hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryUnifiedControl.java

```diff
@@ -183,215 +264,300 @@ void testInitialState() {
+          .get(20, TimeUnit.SECONDS);
+
+      // Allow time for all queued events to be processed
+      Thread.sleep(3000);
+
+      // Collect and analyze results
+      long successResultCount = 0;
+      long retryLaterCount = 0;
+      long maxRetriesCount = 0;
+
+      for (CompletableFuture<ReconTaskController.ReInitializationResult> future : futures) {
+        ReconTaskController.ReInitializationResult result = future.get();
+        switch (result) {
+        case SUCCESS:
+          successResultCount++;
+          break;
+        case RETRY_LATER:
+          retryLaterCount++;
+          break;
+        case MAX_RETRIES_EXCEEDED:
+          maxRetriesCount++;
+          break;
+        default:
+          LOG.warn("Unexpected result: {}", result);
+        }
+      }
 
       // Debug output
-      LOG.info("Final counts - Success: {}, Rejected: {}, ClearTable calls: {}, Final state: {}", 
-          successCount.get(), rejectedCount.get(), clearTableCallCount.get(), NSSummaryTask.getRebuildState());
-
-      // Verify results - only one thread should have successfully executed the rebuild
-      assertEquals(1, clearTableCallCount.get(), 
-          "clearNSSummaryTable should only be called once due to unified control");
-      assertEquals(1, successCount.get(), 
-          "Exactly one rebuild should succeed");
-      assertEquals(threadCount - 1, rejectedCount.get(), 
-          "All other rebuilds should be rejected");
+      LOG.info("Test completed - Total queue attempts: {}, Successful queues: {}, " +
+              "Result breakdown: SUCCESS={}, RETRY_LATER={}, MAX_RETRIES={}, " +
+              "ClearTable calls: {}, Max concurrent: {}, Final state: {}",
+          totalQueueAttempts.get(), successfulQueueCount.get(),
+          successResultCount, retryLaterCount, maxRetriesCount,
+          clearTableCallCount.get(), maxConcurrentExecutions.get(),
+          NSSummaryTask.getRebuildState());
+
+      // CRITICAL INVARIANT: No concurrent executions
+      // The queue + async processing ensures sequential (not concurrent) execution
+      assertEquals(1, maxConcurrentExecutions.get(),
+          "Should never have concurrent executions - queue provides serialization");
+
+      // All threads should have attempted to queue
+      assertEquals(threadCount, totalQueueAttempts.get(),
+          "All threads should have attempted to queue events");
+
+      // At least one thread should have successfully queued
+      assertTrue(successfulQueueCount.get() >= 1,
+          "At least one thread should have successfully queued rebuild");
+
+      // Multiple events may be queued and executed sequentially
+      assertTrue(clearTableCallCount.get() >= 1,
+          "At least one rebuild should execute");
+      assertTrue(clearTableCallCount.get() <= successfulQueueCount.get(),
+          "Number of executions should not exceed successfully queued events");
```

Comment:
nit: please use `assertThat(...)` for better message in case of assertion failure.  These calls can even be chained:

```java
      assertThat(clearTableCallCount.get())
          .isPositive()
          .isLessThanEqual(successfulQueueCount.get());
```